### ISSUE TITLE
Tutorial/Instruction request (data parts selector for Backup/Restore)

### ISSUE THREAD

**Issue by LucasBS1** [NeoApplications/Neo-Backup#909]
It's not a bug, but since there is no adequate category...

Very simple, actually:

I made a list of apps using the **PUT** function.

How do I backup **ONLY THE DATA** of these apps ?

Right now, I have to backup everything and the delete the .apks

**Comment by machiav3lli** [comment]
It's still not possible.

@hg42 I guess this is a FR for your implementation.

**Comment by LucasBS1** [comment]
Thank you for addressing this

**Comment by hg42** [comment]
> @hg42 I guess this is a FR for your implementation.

yes, it's already somewhere in my head...

I was basically waiting, that I can reuse the dialog from the other parts...
as far as I remember, it wasn't a Composable, yet, not sure if I revisited it later

**Comment by hg42** [comment]
yeah, the "openDialogCustom" inflation...

It's not the recommended way.
1. because the Composable must know something about the calling site
2. it's an output parameter, while Compose is a functional paradigm, so there are no output parameters, even if Kotlin allows it, it's not matching the design (and this could create multithreading problems)

I postponed the use of the dialog until this is fixed somehow, because each usage would make it more work to cahnge it.

Would you accept, if I would rewrite this the openDialogCustom thingy? it's sistributed all over the source, so there should nothing be going on in parallel

**Comment by hg42** [comment]
there's more to it...

we have different selections:

apk / data / both
apk / data / dedata / ext / obb / media
no_backup
cache

I think these should all have the same selection:
apk / data / dedata / ext / obb / media / no_backup / cache

Backup page is apk / data basically because with user apps you want apk, but with system apps you don't.
For really individual batches you would want all switches.
So either you have all switches on each entry, or you do the selection group wise, which is more convenient.

Such a dialog does not exist, so I guess I need to create my own.
However, this is not sufficient, because no_backup and cache are global options.

**Comment by hg42** [comment]
will probably be added to next release

**Comment by machiav3lli** [comment]
> I was basically waiting, that I can reuse the dialog from the other parts...
as far as I remember, it wasn't a Composable, yet, not sure if I revisited it later

All dialogs are composables since a while (if I'm not mixing projects).

> Would you accept, if I would rewrite the openDialogCustom thingy to use callbacks? it's distributed all over the source, so there should nothing be going on in parallel.

You can surely test alternatives, but am not sure if it'll replace current components.

> I think these should all have the same selection:
apk / data / dedata / ext / obb / media / no_backup / cache

Not really, all but `no_backup` & `cache` are REAL polytomic types, while these two are *kinda* specific file types acrross these types. Therefore a better differentiation should take place.
